plugins/obs-qsv11/obs-qsv265.c

```diff
@@ -0,0 +1,1118 @@
+			if (la_depth == 0 || la_depth >= 15)
+				obs_data_set_string(settings, "latency",
+						    "normal");
+			else
+				obs_data_set_string(settings, "latency", "low");
+		} else {
+			if (async_depth != 1)
+				obs_data_set_string(settings, "latency",
+						    "normal");
+			else
+				obs_data_set_string(settings, "latency",
+						    "ultra-low");
+		}
+	}
+
+	return true;
+}
+
+static bool update_enhancements(obs_data_t *settings)
+{
+	bool update = false;
+	bool mbbrc = true;
+	if (obs_data_item_byname(settings, "mbbrc") != NULL) {
+		mbbrc = (bool)obs_data_get_bool(settings, "mbbrc");
+		obs_data_erase(settings, "mbbrc");
+		update = true;
+	}
+
+	bool cqm = false;
+	if (obs_data_item_byname(settings, "CQM") != NULL) {
+		cqm = (bool)obs_data_get_bool(settings, "CQM");
+		obs_data_erase(settings, "CQM");
+		update = true;
+	}
+
+	if (update) {
+		bool enabled = (mbbrc && cqm);
+		obs_data_set_bool(settings, "enhancements", enabled);
+	}
+
+	return true;
+}
+
+static bool rate_control_modified(obs_properties_t *ppts, obs_property_t *p,
+				  obs_data_t *settings)
+{
+	const char *rate_control =
+		obs_data_get_string(settings, "rate_control");
+
+	bool bVisible = astrcmpi(rate_control, "VCM") == 0 ||
+			astrcmpi(rate_control, "VBR") == 0;
+	p = obs_properties_get(ppts, "max_bitrate");
+	obs_property_set_visible(p, bVisible);
+
+	bVisible = astrcmpi(rate_control, "CQP") == 0 ||
+		   astrcmpi(rate_control, "LA_ICQ") == 0 ||
+		   astrcmpi(rate_control, "ICQ") == 0;
+	p = obs_properties_get(ppts, "bitrate");
+	obs_property_set_visible(p, !bVisible);
+
+	bVisible = astrcmpi(rate_control, "AVBR") == 0;
+	p = obs_properties_get(ppts, "accuracy");
+	obs_property_set_visible(p, bVisible);
+	p = obs_properties_get(ppts, "convergence");
+	obs_property_set_visible(p, bVisible);
+
+	bVisible = astrcmpi(rate_control, "CQP") == 0;
+	p = obs_properties_get(ppts, "qpi");
+	obs_property_set_visible(p, bVisible);
+	p = obs_properties_get(ppts, "qpb");
+	obs_property_set_visible(p, bVisible);
+	p = obs_properties_get(ppts, "qpp");
+	obs_property_set_visible(p, bVisible);
+
+	bVisible = astrcmpi(rate_control, "ICQ") == 0 ||
+		   astrcmpi(rate_control, "LA_ICQ") == 0;
+	p = obs_properties_get(ppts, "icq_quality");
+	obs_property_set_visible(p, bVisible);
+
+	bVisible = astrcmpi(rate_control, "CBR") == 0 ||
+		   astrcmpi(rate_control, "VBR") == 0;
+	p = obs_properties_get(ppts, "enhancements");
+	obs_property_set_visible(p, bVisible);
+
+	update_latency(settings);
+	update_enhancements(settings);
+
+	return true;
+}
+
+static bool profile_modified(obs_properties_t *ppts, obs_property_t *p,
+			     obs_data_t *settings)
+{
+	// TODO: change logic depending on the selected profile and bit depth
+	const char *profile = obs_data_get_string(settings, "profile");
+	enum qsv_cpu_platform plat = qsv_get_cpu_platform();
+	bool bHdrVisible = ((astrcmpi(profile, "Main 10") == 0) &&
+			    (plat >= QSV_CPU_PLATFORM_ICL));
+	p = obs_properties_get(ppts, "HDR");
```

Comment:
Not sure what this does, or why it's necessary.

Reply:
This is artifact of implementation, where we disable HDR support if no hardware encoder found.

Reply:
Looks like this is gone now.